app/frontend/util/ChampaignAPI.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import _ from 'lodash';
+import $ from 'jquery';
+import { camelizeKeys } from './util';
+
+export type OperationResponse = {
+  success: boolean,
+  errors: ?{[id:string]: string[]}
+}
+
+const parseResponse = (response):OperationResponse => {
```

Comment:
`parseResponse` does some extra work.
My idea is that the code that uses the `createCall` function doesn't need to know about return codes or exceptions. All the logic around HTTP requests is handled here. We just send a simple response object with `success` and `errors` fields.